File: DSA/TEST_CASES/GO_Lang_TestCases/BinarySearch.go
package main

import "fmt"

func search(nums []int, target int) int {
	start, end := 0, len(nums)-1

	for start <= end {

		mid := (end + start) / 2

		if nums[mid] == target {
			return mid
		} else if nums[mid] > target {
			end = mid - 1
		} else if nums[mid] < target {
			start = mid + 1
		}

	}
	return -1
}

func main() {
	nums := []int{-1, 2, 6, 8, 9}
	fmt.Println(search(nums, -1))
	fmt.Println(search(nums, 1))
	fmt.Println(search(nums, 2))
	fmt.Println(search(nums, 8))
	fmt.Println(search(nums, 10))
}

File: DSA/TEST_CASES/Python_TestCases/BinarySearch.py
class Solution:
   
    def search(self, nums, target ):
        start = 0
        end = len(nums) - 1
        while start <= end:
            mid = (start + end) // 2
            if nums[mid] < target:
                start = mid + 1
            elif nums[mid] > target:
                end = mid - 1
            else:
                return mid
        return -1
    
nums =[-1,2,4,5,6,7,8]
S = Solution()
print(S.search(nums,4))
print(S.search(nums,0))
print(S.search(nums,8))
print(S.search(nums,10))